Net07.HW/Net07.WorkWithDate/Net07.WorkWithDate/Program.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+
+namespace Net07.WorkWithDate
+{
+    class Program
+    {
+        enum DaysInMonths
+        {
+            jan = 31,
+            lFeb = 29,
+            Feb = 28,
+            mar = 31,
+            apr = 30,
+            may = 31,
+            jun = 30,
+            jul = 31,
+            aug = 31,
+            sep = 30,
+            oct = 31,
+            nov = 30,
+            dec = 31
+        }
+        enum DaysOfWeek
+        {
+            Monday,
+            Tuesday,
+            Wednsday,
+            Thursday,
+            Friday,
+            Saturday,
+            Sunday
+        }
+        static void Main(string[] args)
+        {
+            GetDayOfWeekEasyLevel();
+            GetDayOfWeekMiddleLevel();
+            GetDayOfWeekHardLevel();
+        }
+
+        private static void GetDayOfWeekHardLevel()
+        {
+            Console.WriteLine("Введите дату ");
+            string dateStr = Console.ReadLine();
+            string[] strArr = dateStr.Split('.');
+            int dayInDate = int.Parse(strArr[0]), monthInDate = int.Parse(strArr[1]), yearInDate = int.Parse(strArr[2]), leapDays = CalcLeapDays(yearInDate);
+            int totalDays = 365 * (yearInDate - 1) + leapDays;
+            for (int i = 1; i < monthInDate; i++)
+                totalDays += GetDaysByMonth(i, Isleap(yearInDate));
+            totalDays += dayInDate;
+            Console.WriteLine(GetDayOfWeekByMod(totalDays));
+            Console.WriteLine(totalDays);
+        }
+
+        private static int CalcLeapDays(int yearInDate)
+        {
+            bool currentYearIsLeap = Isleap(yearInDate);
+            int leapDays = yearInDate / 4;
+            if (currentYearIsLeap)
+                leapDays--;
+            int everyCentury = yearInDate / 100;
+            int everyFourCentury = yearInDate / 400;
+            return leapDays - everyCentury + everyFourCentury;
+        }
+
+        private static bool Isleap(int yearInDate)
+        {
+            if (yearInDate % 4 == 0)
+                return true;
+            else
+                return false;
+        }
+
+        private static DaysOfWeek GetDayOfWeekByMod(int totalDays)
+        {
+            int mod = totalDays % 7;
+            switch(mod)
+            {
+                case 1: return DaysOfWeek.Monday;
+                case 2: return DaysOfWeek.Tuesday;
+                case 3: return DaysOfWeek.Wednsday;
+                case 4: return DaysOfWeek.Thursday;
+                case 5: return DaysOfWeek.Friday;
+                case 6: return DaysOfWeek.Saturday;
+                case 0: return DaysOfWeek.Sunday;
+                default: throw new Exception();
```

Comment:
That's really nice👍
But never throw an exception without a message - exception should always reflect the reason why it has been thrown
```suggestion
                default: throw new Exception("Unexpected day of week");
```

Reply:
fixed